dhall/tests/Dhall/Test/QuickCheck.hs

```diff
@@ -178,7 +178,10 @@ instance Arbitrary Directory where
 instance (Arbitrary s, Arbitrary a) => Arbitrary (Expr s a) where
     arbitrary =
         Test.QuickCheck.suchThat
-            (Generic.Random.genericArbitraryRecG customGens weights)
+            (Generic.Random.withBaseCase
```

Comment:
I believe this is good, but I don't actually quite understand why. Can you help me?

Reply:
This fixes an infinite loop.  Adding one more constructor caused the weights to generate an infinitely growing expression tree.  Using `withBaseCase` ensures that the random generator can still cap the recursion depth when this happens so that we don't have to be so diligent with weights.

Reply:
Oh! This sounds like a bug in `generic-random`. Should we report this?

Reply:
I'm not sure it is a bug.  The documentation from `withBaseCase` seems to suggest that this behavior is expected if you don't use `withBaseCase`